content/blog/debug-cran-additional.md

```diff
@@ -0,0 +1,384 @@
+---
+title: "Debugging and Fixing CRAN's 'Additional Checks' errors"
+date: 2020-01-16
+tags:
+ - dde
+ - R
+---
+
+R packages that are published on CRAN are tested every night on a variety of platforms and on the development version of R to ensure that they continue to work.  In addition, packages that contain compiled code (C, C++ or Fortran) are put through a raft of additional checks to ensure that the compiled code will not cause R to crash.  Once an issue is found, the package maintainer gets an email and usually a fairly short window to fix the package before it is removed from CRAN.  However, replicating the error locally can require installation of all sorts of esoteric tools (and a copy of the development version of R from source) and it's not always obvious to start.
+
+This blog post documents the process I used in clearing three issues from our [`dde`](https://mrc-ide.github.io/dde/) package (which implements a simple solver for delay differential equations - the astute blog reader may recognise it from [previous debugging efforts](https://reside-ic.github.io/blog/debugging-at-the-edge-of-reason/)).
+
+Package authors (or at least I) typically find out about these problems when getting an email from CRAN but they may have been live for some time and are listed as an "Additional issues" section on a packages [check page](https://cran.r-project.org/web/checks/check_results_dde.html) - however, this is shown only when there is a problem!
+
```

Comment:
packages -> either package or package's
and (optionally) - perhaps listed *in* an "Additional issues" section.